src/gen9_avc_encoder.c

```diff
@@ -2595,26 +2865,118 @@ gen9_avc_send_surface_brc_frame_update(VADriverContextP ctx,
+                                0,
+                                avc_ctx->res_brc_image_state_write_buffer.size,
+                                0,
+                                GEN9_AVC_FRAME_BRC_UPDATE_IMAGE_STATE_WRITE_INDEX);
+    if (avc_state->mbenc_brc_buffer_size > 0) {
+        gen9_add_buffer_gpe_surface(ctx,
+                                    gpe_context,
+                                    &(avc_ctx->res_mbenc_brc_buffer),
+                                    0,
+                                    avc_ctx->res_mbenc_brc_buffer.size,
+                                    0,
+                                    GEN95_AVC_FRAME_BRC_UPDATE_MBENC_CURBE_WRITE_INDEX);
+    } else {
+        /*  Mbenc curbe input buffer */
+        gen9_add_dri_buffer_gpe_surface(ctx,
+                                        gpe_context,
+                                        gpe_context_mbenc->dynamic_state.bo,
+                                        0,
+                                        ALIGN(gpe_context_mbenc->curbe.length, 64),
+                                        gpe_context_mbenc->curbe.offset,
+                                        GEN9_AVC_FRAME_BRC_UPDATE_MBENC_CURBE_READ_INDEX);
+        /* Mbenc curbe output buffer */
+        gen9_add_dri_buffer_gpe_surface(ctx,
+                                        gpe_context,
+                                        gpe_context_mbenc->dynamic_state.bo,
+                                        0,
+                                        ALIGN(gpe_context_mbenc->curbe.length, 64),
+                                        gpe_context_mbenc->curbe.offset,
+                                        GEN9_AVC_FRAME_BRC_UPDATE_MBENC_CURBE_WRITE_INDEX);
+    }
+
+    /* AVC_ME Distortion 2D surface buffer,input/output. is it res_brc_dist_data_surface*/
+    gen9_add_buffer_2d_gpe_surface(ctx,
+                                   gpe_context,
+                                   &avc_ctx->res_brc_dist_data_surface,
+                                   1,
+                                   I965_SURFACEFORMAT_R8_UNORM,
+                                   GEN9_AVC_FRAME_BRC_UPDATE_DISTORTION_INDEX);
+
+    /* BRC const data 2D surface buffer */
+    gen9_add_buffer_2d_gpe_surface(ctx,
+                                   gpe_context,
+                                   &avc_ctx->res_brc_const_data_buffer,
+                                   1,
+                                   I965_SURFACEFORMAT_R8_UNORM,
+                                   GEN9_AVC_FRAME_BRC_UPDATE_CONSTANT_DATA_INDEX);
+
+
+    /* MB statistical data surface*/
+    gen9_add_buffer_2d_gpe_surface(ctx,
+                                   gpe_context,
+                                   &avc_ctx->res_mbbrc_mb_qp_data_surface,
+                                   1,
+                                   I965_SURFACEFORMAT_R8_UNORM,
+                                   GEN9_AVC_FRAME_BRC_UPDATE_MB_STATUS_INDEX);
+
+    return;
+}
```

Comment:
it looks gen8_avc_send_surface_brc_frame_update is not necessary and the better way is to reuse the function gen9_avc_send_surface_brc_frame_update.